test/core/rules/test_cases/L043.yml

```diff
@@ -0,0 +1,287 @@
+rule: L043
```

Comment:
I tried to be very thorough about test cases, since this rule has to deal with many features of SQL queries (and how they interact). As you review, it would be 💯 if you could scan over this and see if I have missed any important cases.